### PR TITLE
chore(weave): UI-Side Query Migration Page 3/6: OpVersionPage

### PR DESCRIPTION
**Pull request by tssweeney**
Builds off of https://github.com/wandb/weave/pull/1200 - see that for details

Part 3/6 of the saga: OpVersionPage

This one was fairly straight forward - just adjusting the callsites and filling out `useOpVersions`

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx [hunk 1]
```diff
@@ -16,256 +15,120 @@ import {
...
-            //     overflow: 'hidden',
-            //     pt: 4,
-            //   }}>
             <OpCodeViewer
               entity={entity}
               project={project}
-              opName={opName}
-              opVersions={opVersions.slice().reverse()} // put in increasing order
+              opName={opId}
+              opVersions={opVersions.result ?? []} // put in increasing order
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx:120]: See the comment - the diff viewer is expecting versions in increasing order, so if you remove the reverse here you'll break it.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts [hunk 2]
```diff
@@ -297,7 +416,17 @@ const typeNameToCategory = (typeName: string): ObjectCategory | null => {
   return null;
 };
 
-const objectArtifactVersionNodeToDictNode = (
+const opNameToCategory = (opName: string): OpCategory | null => {
+  const categories = ['train', 'predict', 'score', 'evaluate', 'tune'];
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts:420]: I think we should move this up and rework `type OpCategory` so there's a single source of truth.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx:120]: On this one, the data returned from the server is already in the correct order for display. I manually verified as well.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts (hunk 2 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts:420]: 👍

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx:120]: ok, scratch that. I thought you wanted the most recent/current on left side. Switching back

Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx:120]: 👍

Comment (jamie-rasmussen) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx]: You are changing the order from what we have in prod.

This PR:
Screenshot 2024-02-08 at 10 13 36 AM
Screenshot 2024-02-08 at 10 13 31 AM

Prod:
Screenshot 2024-02-08 at 10 13 21 AM
Screenshot 2024-02-08 at 10 13 15 AM

We could reasonably decide that the selector should  be date decreasing, but I definitely feel like the proper order for the diff is older version on the left, and the previous/next buttons wouldn't do what they say after this either.

Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/OpVersionPage.tsx]: 👍 thanks for the details - I switched it back - my mistake

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts [hunk 3]
```diff
@@ -35,6 +35,17 @@ import {Call as CallTreeSpan} from '../../../Browse2/callTree';
 import {useRuns} from '../../../Browse2/callTreeHooks';
 import {PROJECT_CALL_STREAM_NAME, WANDB_ARTIFACT_REF_PREFIX} from './constants';
 
+const OP_CATEGORIES = [
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts:38]: I think these should be e.g.:
`const OP_CATEGORIES = ['train', ...] as const`

**Review by jamie-rasmussen — APPROVED** [review]

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts (hunk 3 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts:38]: 👍

**Pull request merged**